tasks/main.yml

```diff
@@ -1 +1,31 @@
-# role tasks
+- block:
+    - name: install IPA client packages
+      package:
+        name: "{{ register_idm_packages }}"
+        state: installed
+
+    - name: run the IPA uninstaller, if necessary
+      command: "{{ register_idm_client_base_command }} --uninstall"
+      when: register_idm_client_reconfigure
+
+    - name: run the IPA client installer
+      command: >-
+        {{ register_idm_client_base_command }}
+        --force-join
+        --realm {{ register_idm_server_realm }}
+        --domain {{ register_idm_server_domain }}
+        --principal {{ register_idm_server_admin_username
+            }}@{{ register_idm_server_realm }}
+        --password {{ register_idm_server_admin_password }}
+        --hostname={{ ansible_fqdn }}
```

Comment:
I think it's possible, even reasonable, that this value would be different from ansible_fqdn, since the name in ansible_fqdn isn't guaranteed to be its canonical hostname according to idm/ipa/kerberos/whatever (e.g. if I have an entry in my hosts file, or there's internal/external DNS going on, or other things). There should probably also be a `register_idm_server_hostname`, defaulting to `ansible_fqdn`.

Reply:
I considered this, as well. I'm going to go with introducing that value.